packages/react-liveness/src/components/FaceLivenessDetector/LivenessCheck/LivenessCameraModule.tsx

```diff
@@ -194,10 +209,10 @@ export const LivenessCameraModule = (
       darkModePreference.removeEventListener('change', updateColorModeHandler);
       lightModePreference.addEventListener('change', updateColorModeHandler);
     };
-  }, [canvasRef, videoRef, videoStream, isStartView]);
+  }, [canvasRef, videoRef, videoStream, isStartView, isMetadataLoaded]);
 
   React.useLayoutEffect(() => {
-    if (isCameraReady) {
+    if (isCameraReady && isMetadataLoaded) {
```

Comment:
Using `isMetadataLoaded` to gate behavior in three different side effects creates complexity in readability and refactoring, can we isolate some of this behavior?

Reply:
I think that would be hard to do without a lot of refactoring. But I was actually able to remove this particular gate